Hazel/src/Hazel/Renderer/OrthographicCameraController.cpp

```diff
@@ -62,18 +75,34 @@ namespace Hazel {
 		dispatcher.Dispatch<WindowResizeEvent>(HZ_BIND_EVENT_FN(OrthographicCameraController::OnWindowResized));
 	}
 
-	bool OrthographicCameraController::OnMouseScrolled(MouseScrolledEvent& e)
+	void OrthographicCameraController::SetZoomLevel(float level) 
+	{
+		// translation speed should scale relative to the zoom level changes;
+		float diff = level / m_ZoomLevel;
+		m_CameraTranslationSpeed *= diff;
+		m_ZoomLevel = level;
+		UpdateProjectionMatrix();
+	}
+
+	void OrthographicCameraController::UpdateProjectionMatrix()
```

Comment:
This function could be an inline but all in all great work I like the idea of function mapping like many of the big game engines have!